File: src/main/java/akostenko/aicars/track/StraightTrack.java
package akostenko.aicars.track;

import static java.lang.Math.PI;

import java.util.List;

public class StraightTrack extends Track {

    static final String NAME = "Straight";
    private final int width = 16; // m

    private List<TrackSection> sections = TrackBuilder.start(0,0, - PI / 4, width)
            .straight(1000)
            .done();

    @Override
    public String getTitle() {
        return NAME;
    }

    @Override
    public double getWidth() {
        return width;
    }

    @Override
    public List<TrackSection> sections() {
        return sections;
    }
}

File: src/main/java/akostenko/aicars/track/StraightTrack.kt
package akostenko.aicars.track

import java.lang.Math.PI

class StraightTrack : Track() {
    override val width = 16.0 // m

    override val sections = TrackBuilder.start(0.0, 0.0, -PI / 4, width.toDouble())
            .straight(1000.0)
            .done()

    override val title: String = NAME

    companion object {

        internal val NAME = "Straight"
    }
}
